### PR TITLE
chore: updates to changelog before v1.0.0-rc.1 release

### PR DESCRIPTION
**Pull request by andynog**
Partially close: #3255

This PR updates the `CHANGELOG.md` file with the unreleased information updated before the `v1.0.0-rc.1` release cut (not that there's no `v1.0.0-rc.1` release in the changelog, this will happen only when we cut `v1.0.0` major release, no `unclog release` command was executed)

The changelog was updated using the command:
```
unclog build -a > CHANGELOG.md 
```

This PR also includes a change in the unclog `config.toml` to sort entries by `id` since this seem to be the old behavior and it's easier to diff the file with the historical `CHANGELOG.md` (otherwise it would be sorted alphabetically and this change in ordering for the past releases which is can be very confusing to diff and compare).

#### PR checklist

- [ ] Tests written/updated
- [X] Changelog entry added in `.changelog` (we use [unclog](https://github.com/informalsystems/unclog) to manage our changelog)
- [ ] Updated relevant documentation (`docs/` or `spec/`) and code comments
- [X] Title follows the [Conventional Commits](https://www.conventionalcommits.org/en/v1.0.0/) spec

### CODE REVIEW

File: .changelog/config.toml [hunk 1]
```diff
@@ -14,4 +14,4 @@ release_date_formats = [
 ]
 
 [change_set_sections]
-sort_entries_by = "entry-text"
+sort_entries_by = "id"
```
Comment (melekes) [inline .changelog/config.toml:17]: I'm curious why this was changed.

**Review by melekes — APPROVED** [review]
👍

File: .changelog/config.toml (hunk 1 above)
Comment (andynog) [reply .changelog/config.toml:17]: I explained above, I think this change was added and the changelog was not re-generated after. It messes up with the current ordering, in the past I believe it was the id which makes more sense from a chronological perspective.

**Review by cason — COMMENTED** [review]
I don't have much background to understand this, but suggestions:

- Could we have a Table of Contents?
- Shouldn't updates be classified by "big" releases?
- Why unreleased stuff is on the changelog?

File: CHANGELOG.md [hunk 2]
```diff
@@ -183,6 +165,33 @@
...
+  ([\#2892](https://github.com/cometbft/cometbft/pull/2892))
+- [`config`] Merge `timeout_prevote` and `timeout_precommit`,
+  `timeout_prevote_delta` and `timeout_precommit_delta` into `timeout_round`
+  and `timeout_round_delta` accordingly
+  ([\#2895](https://github.com/cometbft/cometbft/pull/2895))
+`[abci/client]` Deprecate `SetResponseCallback(cb Callback)` in the `Client` interface as it is no
+longer used. ([\#3084](https://github.com/cometbft/cometbft/issues/3084))
+`[mempool]` Change the signature of `CheckTx` in the `Mempool` interface to
+`CheckTx(tx types.Tx, sender p2p.ID) (*abcicli.ReqRes, error)`.
+([\#1010](https://github.com/cometbft/cometbft/issues/1010), [\#3084](https://github.com/cometbft/cometbft/issues/3084))
```
Comment (sergio-mena) [inline CHANGELOG.md:194]: The original file in `.changelog` dir has to be correctly formatted, and `unclog` called again.

File: CHANGELOG.md [hunk 3]
```diff
@@ -291,17 +320,34 @@
 - `[store]` When pruning force compaction of the database. ([\#1972](https://github.com/cometbft/cometbft/pull/1972))
 - `[metrics]` Added metrics to monitor state store access. ([\#1974](https://github.com/cometbft/cometbft/pull/1974))
 - `[metrics]` Added metrics to monitor block store access. ([\#1974](https://github.com/cometbft/cometbft/pull/1974))
+`[config]` Removed unused `[mempool.max_batch_bytes]` mempool parameter
+ ([\#2056](https://github.com/cometbft/cometbft/pull/2056/))
```
Comment (sergio-mena) [inline CHANGELOG.md:324]: need to fix format in original `.changelog` file

File: CHANGELOG.md [hunk 4]
```diff
@@ -489,6 +598,56 @@ Also fixes a small bug in the mempool for an experimental feature.
...
+*November 27, 2023*
+
+Fixes a small bug in the mempool for an experimental feature.
+
+### BUG FIXES
+
+- `[mempool]` Avoid infinite wait in transaction sending routine when
+  using experimental parameters to limiting transaction gossiping to peers
+  ([\#1654](https://github.com/cometbft/cometbft/pull/1654))
+
```
Comment (sergio-mena) [inline CHANGELOG.md:650]: Why does this appear in `main`?

File: CHANGELOG.md [hunk 5]
```diff
@@ -519,6 +678,78 @@ gossip.
   ([\#1558](https://github.com/cometbft/cometbft/pull/1558))
   ([\#1584](https://github.com/cometbft/cometbft/pull/1584))
 
+## v0.37.3
+
```
Comment (sergio-mena) [inline CHANGELOG.md:682]: Same for this hunk, why are these changes here in `v1.x`?

File: CHANGELOG.md (hunk 4 above)
Comment (andynog) [reply CHANGELOG.md:650]: will fix it as per our conversation

File: CHANGELOG.md (hunk 5 above)
Comment (andynog) [reply CHANGELOG.md:682]: will fix it as per our conversation

File: CHANGELOG.md (hunk 3 above)
Comment (andynog) [reply CHANGELOG.md]: fixed

File: CHANGELOG.md (hunk 2 above)
Comment (andynog) [reply CHANGELOG.md]: fixed

**Comment by andynog** [comment]
@sergio-mena I've added the changes requested. In the last commits I've added these changes:
- Re-added the `summary.md` doc.
- Removed releases that should not be in the `.changelog` folder. I've ensured that these entries matched the entries in their respective releases (folder diff) so we are not loosing anything.
- Many formatting fixes and adding information that was missing to ensure entries are consistent and accurate.
- Re-generated the `CHANGELOG.md` file after these changes were in place.

File: .changelog/unreleased/summary.md [hunk 6]
```diff
@@ -0,0 +1,48 @@
+*December 4, 2023*
```
Comment (melekes) [inline .changelog/unreleased/summary.md:1]:
```suggestion
*June 20, 2024*
```

File: CHANGELOG.md [hunk 7]
```diff
@@ -2,22 +2,196 @@
 
 ## Unreleased
 
+*December 4, 2023*
```
Comment (melekes) [inline CHANGELOG.md:5]:
```suggestion
*June 20, 2024*
```

**Review by melekes — APPROVED** [review]
👍

File: .changelog/unreleased/summary.md [hunk 8]
```diff
@@ -0,0 +1,48 @@
...
+6. Moving many Go packages that are currently publicly accessible into the
+   `internal` directory such that the team can roll out substantial changes in
+   future without needing to worry about causing breakages in users' codebases.
+   The massive surface area of previous versions has in the past significantly
+   hampered the team's ability to roll out impactful new changes to users, as
+   previously such changes required a new breaking release (which currently
+   takes 6 to 12 months to reach production use for many users). See [ADR
+   109](./docs/architecture/adr-109-reduce-go-api-surface.md) for more details.
+7. Proposer-Based Timestamps (PBTS) support. PBTS is a Byzantine fault-tolerant
+algorithm used by CometBFT for computing block times. See [spec](./spec/consensus/proposer-based-timestamp) doc for PBTS.
```
Comment (sergio-mena) [inline .changelog/unreleased/summary.md:38]:
```suggestion
7. Proposer-Based Timestamps (PBTS) support. PBTS is a Byzantine fault-tolerant
    algorithm used by CometBFT for computing block times.
    When activated on a chain, it replaces the pre-existing BFT-time algorithm.
    See [spec](./spec/consensus/proposer-based-timestamp) doc for PBTS.
```

Thanks for adding this @andynog !

File: .changelog/unreleased/summary.md [hunk 9]
```diff
@@ -0,0 +1,48 @@
...
+
+None of these changes are state machine-breaking for CometBFT-based networks,
+but could be breaking for some users who depend on the Protobuf definitions type
+URLs. See the [upgrading guidelines](./UPGRADING.md) and specific changes below
+for more details.
+
+**NB: This version is still an alpha-series release, which means that
```
Comment (sergio-mena) [inline .changelog/unreleased/summary.md:45]:
```suggestion
**NB: This version is still a release candidate, which means that
```

File: .changelog/unreleased/summary.md [hunk 10]
```diff
@@ -0,0 +1,48 @@
...
+API-breaking changes might still be introduced until such time that a _release
```
Comment (sergio-mena) [inline .changelog/unreleased/summary.md:46]:
```suggestion
API-breaking changes, although very unlikely, might still be introduced until such time that a _release
```

File: .changelog/unreleased/summary.md [hunk 11]
```diff
@@ -0,0 +1,48 @@
...
+candidate_ is cut.** See [RELEASES.md](./RELEASES.md) for more information on
```
Comment (sergio-mena) [inline .changelog/unreleased/summary.md:47]:
```suggestion
API-breaking changes, although very unlikely, might still be introduced
before the final release.** See [RELEASES.md](./RELEASES.md) for more information on
```

File: .changelog/unreleased/summary.md (hunk 9 above)
Comment (andynog) [reply .changelog/unreleased/summary.md:47]: good catch! Thanks !
